src/routes/+layout.svelte

```diff
@@ -82,14 +89,42 @@
                 loading.set(false);
             }
         }
+
+        /**
+         * typesafe-i18n here
+         */
+
+        const detectedLocale = detectLocale(localStorageDetector);
+        await chooseLocale(detectedLocale);
+        // console.log($LL.startup());
+        localeToSelect = $locale;
     });
 
+    const chooseLocale = async (locale: Locales) => {
+        await loadLocaleAsync(locale);
+        setLocale(locale);
+    };
+
+    let localeToSelect: Locales;
+    $: localeToSelect && chooseLocale(localeToSelect);
+
+    $: $locale && localStorage.setItem('lang', window.navigator.language);
```

Comment:
Won't this reset the user set language?

Reply:
Also, maybe all the locale logic can be kept in LanguageSelector? Since it's called only once

Reply:
changes done.